lib/services/devices/deviceService.js

```diff
@@ -577,9 +577,17 @@ function findOrCreate(deviceId, group, callback) {
             if ('expressionLanguage' in group && group.expressionLanguage !== undefined) {
                 newDevice.expressionLanguage = group.expressionLanguage;
             }
-            registerDevice(newDevice, function (error, device) {
-                callback(error, device, group);
-            });
+            // Check autoprovision flag in order to register or not device
+            logger.info(context, 'newDevice %j group %j', newDevice, group);
+            if (newDevice.autoprovision === undefined || newDevice.autoprovision === true) {
+                registerDevice(newDevice, function (error, device) {
+                    callback(error, device, group);
+                });
+            } else {
+                logger.info(context, 'Device %j not provisioned due autoprovision is disabled by its conf', newDevice);
+                callback(null, newDevice, group);
+                callback(new errors.DeviceNotFound(deviceId));
```

Comment:
Check if this could be a problem from the point of view of backward compatibility on existing cases.

Reply:
Check:

```
> use iotagent_json_default
switched to db iotagent_json_default
> db.devices.count({autoprovision: {$exists: true}})
0
> use iotagent_json_default
switched to db iotagent_json_default
> db.devices.count({autoprovision: {$exists: true}})
0
> db.groups.count({autoprovision: {$exists: true}})
0
> use iotagent_ul_default
switched to db iotagent_ul_default
> db.devices.count({autoprovision: {$exists: true}})
0
> db.groups.count({autoprovision: {$exists: true}})
0
```

So everything ok and NTC